src/server/handlers/launchpad.js

```diff
@@ -387,5 +383,6 @@ export const requestSnapBuilds = (req, res) => {
         }
       });
     })
-    .catch((error) => responseError(res, error));
+    .catch((error) => prepareError(error))
+    .catch((error) => res.status(error.status).send(error.body));
```

Comment:
I've refactored it a bit.  Is this any better?  I had to extract at least part of the error preparation logic in order that `completeSnapAuthorization` could just throw errors rather than sending a response.